### ISSUE TITLE
nanocolors failing Jest tests

### ISSUE THREAD

**Issue by dan-dm** [ai/nanocolors#10]
Hey, I'm trying to use yours truly, *nanocolors* instead of chalk in a simple JS program and I keep getting this error when running `npm test` on the test.js file 

![image](https://user-images.githubusercontent.com/58673649/134578350-9058fbdb-055d-40b7-b445-a7a90dc6407a.png)

App works perfectly fine when run, ( has `import  { red, blue } from 'nanocolors';` @ the top ) and the class tested is being exported.

Any ideas? Just substituting `nanocolors` imports and syntax to `chalk `makes Jest run the test fine.🙆‍♂️

**Comment by TrySound** [comment]
Which version of jest are you using?

**Comment by dan-dm** [comment]
> Which version of jest are you using?

That would be v27.2.1, latest currently. 
  
```
{
  "name": "jest",
  "description": "Delightful JavaScript Testing.",
  "version": "27.2.1",
  "main": "./build/jest.js",
  "types": "./build/jest.d.ts",
  "exports": {
    ".": "./build/jest.js",
    "./package.json": "./package.json",
    "./bin/jest": "./bin/jest.js"
  }
```

Thanks!

**Comment by ai** [comment]
Very strange. As I know Jest resolver still do not support `package.export`. Could it be a case?

Can you give a repo to reproduce the issue?

**Comment by TrySound** [comment]
@ai Btw node can infer named exports from commonjs when it's each export is explicitly defined for example
```
exports.blue = blue
```
probably `module.exports = { blue }` too but not sure

**Comment by ai** [comment]
@TrySound maybe. But I will need reproduction  repo to be sure that it is a problem before removing this optimization.

**Comment by dan-dm** [comment]
Thank you both, I fixed the issue by changing "type" from "module" to "commonjs" in `package.json` and replacing the `import `statement to the `require()` syntax.

> Very strange. As I know Jest resolver still do not support `package.export`. Could it be a case?

I am using Jest with the experimental modules flag `"test": "node --experimental-vm-modules node_modules/jest/bin/jest.js"` and it was working just fine with chalk with the import syntax, that's what threw me off.

**Comment by ai** [comment]
> Thank you both, I fixed the issue by changing "type" from "module" to "commonjs" in package.json and replacing the import statement to the require() syntax.

We need to find a real source.

> I am using Jest with the experimental modules flag "test": "node --experimental-vm-modules node_modules/jest/bin/jest.js" and it was working just fine with chalk with the import syntax, that's what threw me off.

I do the same for my projects.

Do you use TS? What Jest settings do you have?

**Comment by dan-dm** [comment]
> We need to find a real source.

I agree, but I'm not sure if I can be of much help, I'm a beginner in JS. Maybe I'm doing something wrong?

These are the same JS snippet and test, only one with the chalk import and the other with nanocolors. `node` runs both apps succesfully.

* [daisy-chalk](https://github.com/dan-dm/daisy-chalk)
* [daisy-nanocolors](https://github.com/dan-dm/daisy-nanocolors)

Only did `npm install` and `npm test` on both, and one works while the other doesn't 🙆‍♂️

![daisy-chalk](https://user-images.githubusercontent.com/58673649/134696628-c9ba8bed-0d0d-4051-8cfe-7195c36c1d55.png)

And for chalk:

![nanocolors-chalk](https://user-images.githubusercontent.com/58673649/134696621-3865ea6b-9931-4139-982b-c092fb72227b.png)

> Do you use TS? What Jest settings do you have?
Haven't modified any Jest settings.

**Comment by marco-carvalho** [comment]
i created [this stackblitz](https://stackblitz.com/edit/node-sc2kjj?file=index.test.js) and it's working here

**Comment by ai** [comment]
@dan-dm what Node.js version do you use? Maybe it is Node.js ESM bug (we have some in 13.x).

**Comment by ai** [comment]
> daisy-nanocolors

Thanks. I reproduced the issue. Going to debug it.

**Comment by ai** [comment]
Fixed d7e194b

@TrySound yeap, you were right. The problem was created by 2 issues:

1. Jest resolver uses CJS version.
2. CJS → ESM exports requires explicit export like `module.export = { x }` or `module.export.x = …`

**Comment by ai** [comment]
The fix was released in 0.2.2.

Thanks for the example. It helped a lot.

**Comment by dan-dm** [comment]
You're welcome! Me alegro de que pude ayudar 😊
